docs/custom-image.md

```diff
@@ -1,96 +1,136 @@
-wget https://download.clearlinux.org/current/config/image/cloudguest.yaml
-sed -i '/size: \"864M\"/d' cloudguest.yaml
-sed -i 's/\"800M\"/\"2G\"/g' cloudguest.yaml
-sed -i 's/bootloader,/bootloader,\n    curl,\n    iperf,/g' cloudguest.yaml
-sed -i 's/systemd-networkd-autostart/sysadmin-basic,\n    systemd-networkd-autostart/g' cloudguest.yaml
-# Create the custom cloudguest image
-clr-installer -c cloudguest.yaml
-# Make the guest accessible through ssh
-sudo mkdir -p /etc/ssh
-sudo bash -c "echo 'PermitRootLogin yes' >> /etc/ssh/sshd_config"
+$ qemu-img convert -p -f qcow2 -O raw focal-server-cloudimg-amd64.img focal-server-cloudimg-amd64.raw
 ```
 
-### Retrieve the image
+### Identify the Linux partition
 
-Once the new image has been created and the guest is accessible through
-`ssh`, it is time to retrieve the image from the host.
+The goal is to mount the image rootfs so that it can be modified as needed.
+That's why we need to identify where the Linux filesystem partition is located
+in the image.
 
 ```bash
-# Retrieve new image (this is a raw image)
-scp root@192.168.249.2:cloudguest.img .
-mv cloudguest.img clear-cloudguest-raw.img
-# Create the QCOW image from the RAW image
-qemu-img convert -p -f raw -O qcow2 clear-cloudguest-raw.img clear-cloudguest.img
-# Compress the QCOW image
-xz -k -T $(nproc) clear-cloudguest.img
+$ sudo fdisk -l focal-server-cloudimg-amd64.raw
+Disk focal-server-cloudimg-amd64.raw: 2.2 GiB, 2361393152 bytes, 4612096 sectors
+Units: sectors of 1 * 512 = 512 bytes
+Sector size (logical/physical): 512 bytes / 512 bytes
+I/O size (minimum/optimal): 512 bytes / 512 bytes
+Disklabel type: gpt
+Disk identifier: A1171ABA-2BEA-4218-A467-1B2B607E5953
+
+Device                             Start     End Sectors  Size Type
+focal-server-cloudimg-amd64.raw1  227328 4612062 4384735  2.1G Linux filesystem
+focal-server-cloudimg-amd64.raw14   2048   10239    8192    4M BIOS boot
+focal-server-cloudimg-amd64.raw15  10240  227327  217088  106M EFI System
+
+Partition table entries are not in disk order.
 ```
 
-## Switch CI to use the new image
+### Mount the Linux partition
 
-### Upload to Azure storage
+```bash
+$ mkdir -p /mnt
+$ sudo mount -o loop,offset=$((227328 * 512)) focal-server-cloudimg-amd64.raw /mnt
+```
 
-The next step is to update the image stored as part of the Azure storage
-account, replacing it with the newly created image. This will make this
-new image available from the integration tests.
-This is usually achieved through the web interface.
+### Change root directory
 
-### Update integration tests
+Changing the root directory will allow us to install new packages to the rootfs
+contained by the cloud image.
 
-Last step is about updating the integration tests to work with this new image.
-The key point is to identify the UUID of this new image so that it can be used
-directly from the tests.
+```bash
+$ sudo chroot /mnt
+# mount -t proc proc /proc
+# mount -t devpts devpts /dev/pts
```

Comment:
We're mixing `$` and `#` as our shell prompt. We should pick only one.

Reply:
Well I'm not mixing, I was trying to make the difference between the shell from the host machine before  and after the `chroot`. Any suggestion on how to make it obvious?

Reply:
Oh, I see. Not sure we absolutely need to differentiate, but maybe use a special prompt:

```
[chroot] $ mount -t proc proc /proc
```

Reply:
Sounds good I'll do that.

Reply:
Make sure you keep the `#` as it will be root inside the chroot.

(FWIW I completely disagree with ever prefixing commands like this as it makes it impossible to copy and paste a whole block at once.)

Reply:
The context of these instructions is *us* in 12 months time when we need to build a new image.

Reply:
@rbradford I agree about the "not prefixing", but how can I show the difference between the command to run and the expected output from a command?

Reply:
None of those commands generate output that the user cares about...I think we can recognise what the commands are.